packages/block-directory/src/store/controls.js

```diff
@@ -6,7 +11,7 @@
  * @return {Promise} Promise which will resolve when the asset is loaded.
  */
 export const loadScript = ( asset ) => {
-	if ( ! asset || ! /\.js$/.test( asset ) ) {
+	if ( ! asset || ! /\.js$/.test( getPath( asset ) ) ) {
```

Comment:
Just noting that this will force urls to be absolute based on `getPath`'s implementation.

Reply:
We'll need to add the protocol in the End to End test:

https://github.com/WordPress/gutenberg/blob/a2ed419b1c89ca6e855292810a5eb99a40eae9d8/packages/e2e-tests/specs/editor/plugins/block-directory-add.test.js#L37

Reply:
It should always be a full URL from the API, once https://github.com/WordPress/gutenberg/pull/23499 is merged. This should be fixed in 8274c1c814b6841bfebb2b635409337516f4be95